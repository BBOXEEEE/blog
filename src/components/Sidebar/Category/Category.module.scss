@import "../../../assets/scss/variables";
@import "../../../assets/scss/mixins";

.category {
  @include margin-bottom(1);

  .list {
    list-style: none;
    @include margin-equal(0);
    @include padding-equal(0);

    .item {
      @include margin(0.5, 0);
      @include padding-equal(0);

      .link {
        border: 0;
        color: var(--color-typographic-base-font);
        font-size: $typographic-base-font-size * 0.9;
        font-style: italic;
        font-weight: normal;

        &:hover,
        &:focus {
          border-bottom: 1px solid $color-blue;
          color: $color-blue;
        }

        &.active {
          border-bottom: 1px solid var(--color-typographic-base-font);
          color: var(--color-typographic-base-font);
        }
      }
    }
  }
}

.title {
    color: $color-blue;
    font-size: $typographic-base-font-size * 1.1;
    font-weight: bold;
}
