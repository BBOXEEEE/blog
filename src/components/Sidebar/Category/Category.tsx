import React from "react";

import { Link } from "gatsby";

import * as styles from "./Category.module.scss";

type Props = {
  category: Array<{
    label: string;
    path: string;
  }>;
};

const Category: React.FC<Props> = ({ category }: Props) => (
<>
  <div className={styles.title}>Category</div>
  <nav className={styles.category}>
    <ul className={styles.list}>
      {category?.map((item) => (
        <li className={styles.item} key={item.path}>
          <Link
            to={item.path}
            className={styles.link}
            activeClassName={styles.active}
          >
            {item.label}
          </Link>
        </li>
      ))}
    </ul>
  </nav>
</>
);

export default Category;
