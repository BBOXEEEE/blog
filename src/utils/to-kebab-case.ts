const toKebabCase = (str: string = ""): string =>
  str
    .match(/[\uAC00-\uD7AF]+|[A-Z]{2,}(?=[A-Z][a-z]+[0-9]*|\b)|[A-Z]?[a-z]+[0-9]*|[A-Z]|[0-9]+/g)
    ?.map((word) => word.toLowerCase())
    .join("-") || "";

export default toKebabCase;
