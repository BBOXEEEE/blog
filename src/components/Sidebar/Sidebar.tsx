import React from "react";

import { useSiteMetadata, useCategoriesList } from "@/hooks";

import { Author } from "./Author";
import { Contacts } from "./Contacts";
import { Copyright } from "./Copyright";
import { Menu } from "./Menu";
import { Category } from "./Category";

import * as styles from "./Sidebar.module.scss";

type Props = {
  isIndex?: boolean;
};

const Sidebar = ({ isIndex }: Props) => {
  const { author, copyright, menu } = useSiteMetadata();
  const categoriesList = useCategoriesList();
  
  const category = categoriesList.map(cat => ({
    label: cat.fieldValue,
    path: `/category/${cat.fieldValue.toLowerCase().replace(/\s+/g, '-')}`,
    count : cat.totalCount
  }));

  return (
    <div className={styles.sidebar}>
      <div className={styles.inner}>
        <Author author={author} isIndex={isIndex} />
        <Contacts contacts={author.contacts} />
        <Menu menu={menu} />
        <Category category={category} />
        <Copyright copyright={copyright} />
      </div>
    </div>
  );
};

export default Sidebar;
