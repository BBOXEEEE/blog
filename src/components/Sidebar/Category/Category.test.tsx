import React from "react";

import { Category } from "@/components/Sidebar/Category";
import * as mocks from "@/mocks";
import { testUtils } from "@/utils";

describe("Category", () => {
  test("renders correctly", () => {
    const props = { category: mocks.sidebarCategory };
    const tree = testUtils
      .createSnapshotsRenderer(<Category {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
