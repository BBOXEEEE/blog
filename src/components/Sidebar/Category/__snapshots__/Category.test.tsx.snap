// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category renders correctly 1`] = `
[
  <div>
    <a
      href="/categories"
    >
      Category
    </a>
  </div>,
  <nav>
    <ul>
      <li>
        <a
          href="/category/typography"
        >
          Typography
           (
          1
          )
        </a>
      </li>
    </ul>
  </nav>,
]
`;
